hca/dss/__init__.py

```diff
@@ -269,19 +289,20 @@ def download_manifest_v2(self, manifest, replica, num_retries=10, min_delay_seco
         """
         rows = self._parse_manifest(manifest)
         errors = 0
-        for row in rows:
-            file_uuid = row['file_uuid']
-            sha = row['file_sha256']
-            path = self._file_path(sha)
-            version = row['file_version']
-            size = row['file_size']
-            if os.path.exists(path):
-                logger.info("File %s already present. Skipping download.", file_uuid)
-            else:
-                logger.info("File %s: Retrieving...", file_uuid)
+        if threads is None and (3, 4) <= sys.version_info < (3, 5):
+            # Py3.4 doesn't have a default so we just do it here for portability
+            max_workers = (os.cpu_count() or 1) * 5
+        else:
+            max_workers = threads
+        with concurrent.futures.ThreadPoolExecutor(max_workers=max_workers) as executor:
```

Comment:
Since download is blocked mainly on IO, threads seem the obvious choice. The extra overhead from processes would likely just slow things down.

Reply:
Since that's a guess, I think it might be worthwhile to do testing.

Reply:
Went ahead and tested this.  Seems to take about the same amount of time.  PR seems good for approval.

In case you were interested:
```
(venv) ubuntu@ip-172-31-47-147:~/delete/dcp-cli$ python i.py
Populating manifests.
Writing manifests with 445 files.
Downloading w/Processing.
[wP 2 Processes]: 445 files downloaded in 44.589733839035034 seconds.
[wP 4 Processes]: 445 files downloaded in 26.317150354385376 seconds.
[wP 8 Processes]: 445 files downloaded in 14.801072597503662 seconds.
[wP 16 Processes]: 445 files downloaded in 11.219650745391846 seconds.
[wP 32 Processes]: 445 files downloaded in 12.45693302154541 seconds.
[wP 64 Processes]: 445 files downloaded in 12.688286542892456 seconds.
Downloading w/Threads.
[v2 2 Threads]: 445 files downloaded in 47.60785889625549 seconds.
[v2 4 Threads]: 445 files downloaded in 25.576399564743042 seconds.
[v2 8 Threads]: 445 files downloaded in 14.707956790924072 seconds.
[v2 16 Threads]: 445 files downloaded in 11.791214227676392 seconds.
[v2 32 Threads]: 445 files downloaded in 13.140044212341309 seconds.
[v2 64 Threads]: 445 files downloaded in 13.080667734146118 seconds.
```